include/seqan3/range/view/kmer_hash.hpp

```diff
@@ -61,47 +659,56 @@ struct kmer_hash_fn
-     * | std::ranges::RandomAccessRange  |                                       | *preserved*                                        |
-     * | std::ranges::ContiguousRange    |                                       | *lost*                                             |
-     * |                                 |                                       |                                                    |
-     * | std::ranges::ViewableRange      | *required*                            | *guaranteed*                                       |
-     * | std::ranges::View               |                                       | *guaranteed*                                       |
-     * | std::ranges::SizedRange         |                                       | *preserved*                                        |
-     * | std::ranges::CommonRange        |                                       | *preserved*                                        |
-     * | std::ranges::OutputRange        |                                       | *lost*                                             |
-     * | seqan3::ConstIterableRange      |                                       | *preserved*                                        |
-     * |                                 |                                       |                                                    |
-     * | seqan3::reference_t             | seqan3::Semialphabet                  | std::size_t                                        |
-     *
-     * See the \link view view submodule documentation \endlink for detailed descriptions of the view properties.
-     *
-     * ### Example
-     * \snippet test/snippet/range/view/kmer_hash.cpp usage
-     * \hideinitializer
-     */
-    inline auto constexpr kmer_hash = detail::kmer_hash_fn{};
+/*!\name Alphabet related views
+ * \{
+ */
+
+/*!\brief               Computes hash values for each position of a range via a given shape.
+ * \tparam urng_t       The type of the range being processed. See below for requirements. [template parameter is
+ *                      omitted in pipe notation]
+ * \param[in] urange    The range being processed. [parameter is omitted in pipe notation]
+ * \param[in] shape     The seqan3::shape that determines how to compute the hash value.
+ * \returns             A range of std::size_t where each value is the hash of the resp. k-mer.
+ *                      See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t  | `urng_t` (underlying range type)   | `rrng_t` (returned range type)   |
+ * |---------------------------------|:----------------------------------:|:--------------------------------:|
+ * | std::ranges::InputRange         | *required*                         | *preserved*                      |
```

Comment:
Please add this line at the top:
```
 *  | std::Semiregular                |                                       | *preserved*                            |
 *  |                                 |                                       |                                      |
```

Since not all views are guaranteed to be Semiregular nowadays, we should indicate which are. I will change this later for the existing views.